src/Sarif/HashUtilities.cs

```diff
@@ -2,21 +2,71 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
 using System.IO;
 using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
 
 namespace Microsoft.CodeAnalysis.Sarif
 {
     public static class HashUtilities
     {
+        static HashUtilities() => FileSystem = new FileSystem();
+
+        private static IFileSystem _fileSystem;
+        internal static IFileSystem FileSystem
+        {
+            get
+            {
+                _fileSystem = _fileSystem ?? new FileSystem();
+                return _fileSystem;
+            }
+
+            set
+            {
+                _fileSystem = value;
+            }
+        }
+
+        public static IDictionary<string, HashData> MultithreadedComputeTargetFileHashes(IEnumerable<string> analysisTargets)
+        {
+            if (analysisTargets == null) { return null; }
+
+            Console.WriteLine("Computing file hashes...");
```

Comment:
>Console.WriteLine("Computing file hashes..."); [](start = 12, length = 46)

Did you mean to leave this in? I don't think you want console output from a utility function.

Reply:
It's been useful to understand when hashing is occurring but it isn't strictly necessary. to 'fix' this, i'd need to plumb through a logger object to some relevant caller. I'll go look and see if there's a natural place to do this. Agree it's weird code for a utility.

---
In reply to: [303174902](https://github.com/microsoft/sarif-sdk/pull/1574#discussion_r303174902) [](ancestors = 303174902)